src/viewport.js

```diff
@@ -51,19 +53,23 @@ Viewport.prototype = {
     }
   },
   getState() {
-    const { element, lastY } = this
-    let width, height, y
+    const { element, lastX, lastY } = this
+    let width, height, positionX, positionY
     if (element === window) {
       width = element.innerWidth
       height = element.innerHeight
-      y = element.pageYOffset
+      positionX = element.pageXOffset
+      positionY = element.pageYOffset
     } else {
       width = element.offsetWidth
       height = element.offsetHeight
-      y = element.scrollTop
+      positionY = element.scrollTop
+      positionX = element.scrollLeft
     }
-    const yDirection = lastY < y ? 'down' : 'up'
-    return { width, height, y, yDirection }
+    const directionX = lastX < positionX ? 'right' : 'left'
```

Comment:
Likely need a condition where if last and current are the same, the direction is 'none' or null because you only scroll one axis at a time.